rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -270,19 +270,41 @@ rmw_create_publisher(
     goto fail;
   }
 
+  switch (qos_policy.history) {
+    case RMW_QOS_POLICY_KEEP_LAST_HISTORY:
+      datawriter_qos.history.kind = DDS::KEEP_LAST_HISTORY_QOS;
+      break;
+    case RMW_QOS_POLICY_KEEP_ALL_HISTORY:
+      datawriter_qos.history.kind = DDS::KEEP_ALL_HISTORY_QOS;
+      break;
+    default:
+      throw std::runtime_error("Unknown QoS history policy");
```

Comment:
My previous comment is over a month ago when we did not yet have the error message API in rmw.

Instead of throwing or asserting this should use the following as it is done for all other errors in the rmw implementations:

    RMW_SET_ERROR_MSG("boom");
    goto fail;